src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx

```diff
@@ -1311,4 +1311,12 @@
   <data name="OperatorOverloadsHaveNamedAlternatesCodeFixTitle" xml:space="preserve">
     <value>Add operator overload named alternate</value>
   </data>
-</root>
+  <data name="AvoidMultipleEnumerationsMessage" xml:space="preserve">
+    <value>To prevent unnecessary enumerations of the 'IEnumerable' collection, consider caching it first by calling a method such as 'ToList' or 'ToArray'</value>
```

Comment:
Message still seems partial to me - it directly gets into what needs to be done without telling what is the issue with current code. How about prepending the title, like:

```
Possible multiple enumerations over 'IEnumerable' collection in this method. To prevent unnecessary enumerations, consider caching it first by calling a method such as 'ToList' or 'ToArray'
```

Reply:
Sounds good. I would change "enumerations over" to "enumerations of" though.

Reply:
Change it to 
Possible multiple enumerations of 'IEnumerable' collection in this method. To prevent unnecessary enumerations, consider caching it first by calling a method such as 'ToList' or 'ToArray'